packages/openstef-core/src/openstef_core/feature_engineering/temporal_transforms/holiday_features.py

```diff
@@ -0,0 +1,178 @@
+        """
+        years = range(int(index.year.min()), int(index.year.max()) + 1)  # type: ignore[reportUnknownMemberType]
+        return dict(holidays.country_holidays(self.country_code, categories=["public"], years=years, language="en_US"))
+
+    @staticmethod
+    def _sanitize_holiday_name(holiday_name: str) -> str:
+        """Convert holiday name to a valid Python identifier for feature names.
+
+        Args:
+            holiday_name: Original holiday name from holidays library.
+
+        Returns:
+            Sanitized feature name (lowercase, underscores, no special chars).
+        """
+        # Convert to lowercase and replace spaces/special chars with underscores
+        sanitized = holiday_name.lower()
+        sanitized = "".join(c if c.isalnum() else "_" for c in sanitized)
+        # Remove consecutive underscores and strip leading/trailing underscores
+        return "_".join(filter(None, sanitized.split("_")))
+
+    @staticmethod
+    def _create_general_holiday_feature(index: pd.DatetimeIndex, country_holidays: dict[date, str]) -> pd.Series:
+        """Create the general is_holiday feature.
+
+        Args:
+            index: DatetimeIndex for the dataset.
+            country_holidays: Dictionary of holidays.
+
+        Returns:
+            Series with 1 for holidays, 0 for non-holidays.
+        """
+        holiday_dates: list[date] = list(country_holidays.keys())
+        return pd.Series(index.date, index=index, name="is_holiday").isin(holiday_dates).astype(int)  # type: ignore[reportUnknownMemberType]
+
+    def _create_individual_holiday_features(
+        self, index: pd.DatetimeIndex, country_holidays: dict[date, str]
+    ) -> pd.DataFrame:
+        """Create individual features for each specific holiday.
+
+        Args:
+            index: DatetimeIndex for the dataset.
+            country_holidays: Dictionary mapping dates to holiday names.
+
+        Returns:
+            DataFrame with individual holiday features.
+        """
+        if not self.include_individual_holidays or not country_holidays:
+            return pd.DataFrame(index=index)
+
+        # Get unique holiday names and create sanitized feature names
+        unique_holidays = set(country_holidays.values())
+        holiday_feature_mapping = {holiday: f"is_{self._sanitize_holiday_name(holiday)}" for holiday in unique_holidays}
+
+        # Initialize all holiday features with zeros
+        holiday_features = pd.DataFrame(data=0, index=index, columns=list(holiday_feature_mapping.values()))
+
+        # Set 1 for dates that match each specific holiday
+        for holiday_date, holiday_name in country_holidays.items():
+            feature_name = holiday_feature_mapping[holiday_name]
+            # Use .loc to handle cases where holiday_date might not be in index
+            mask = index.date == holiday_date
+            if mask.any():
+                holiday_features.loc[mask, feature_name] = 1
+
+        return holiday_features
```

Comment:
I wonder if this loop might become a bottleneck in the future and if it can be made efficient. 

Currenltly if N is the length of the data and H is number of holidays, the complexity of this function is O(NH*2).
Perhaps it's possible to make this an O(N+H) by first putting holidays in a sparse dataframe and then merging on index?

Does this also work for holidays that span multiple days?